tests/impl.cpp

```diff
@@ -1656,11 +1656,25 @@ result_t test_mm_extract_pi16(const SSE2NEONTestImpl &impl, uint32_t i)
     // information
 #if defined(__clang__)
     uint64_t *_a = (uint64_t *) impl.mTestIntPointer1;
-    const int imm = 1;
 
     __m64 a = do_mm_load_m64((const int64_t *) _a);
-    int32_t c = _mm_extract_pi16(a, imm);
-    ASSERT_RETURN((uint64_t) c == ((*_a >> ((imm & 0x3) * 16)) & 0xFFFF));
+    int c;
+    switch (i & 0x3) {
+        case 0:
+            c = _mm_extract_pi16(a, 0);
+            break;
+        case 1:
+            c = _mm_extract_pi16(a, 1);
+            break;
+        case 2:
+            c = _mm_extract_pi16(a, 2);
+            break;
+        case 3:
+            c = _mm_extract_pi16(a, 3);
+            break;
+    }
+
+    ASSERT_RETURN((uint64_t) c == ((*_a >> ((i & 0x3) * 16)) & 0xFFFF));
```

Comment:
I suggest you to have a variable `idx` which equals to `i&03`, then you can replace `idx` in the code to make it more readable.